### DISCUSSION TITLE
Bootstrapper and CapsuleContainer

### DISCUSSION THREAD

**Discussion by busslina** [GregoryConrad/rearch-dart#150] [General]
Have you thought about the possibility of allowing external creation of CapsuleContainer and the passing it as an optional parameter to Bootstrapper (Like Riverpod does with UncontrolledProviderContainer).

I'm 95% sure that you will answer that this is not a good idea haha

**Comment by GregoryConrad** [comment]
That’s fine, it’s possible via the CapsuleContainerProvider (which the bootstrapper wraps around).

that being said, it probably shouldn’t be used apart from testing, at least as far as I can think of.

**Reply by busslina** [reply]
Okay, I achieved the bootstrap logic:
- Only needs of a single Component compared with the two needed in Flutter.
- For now, capsuleContainer is declared global in `bootstrap.dart`, just for migrating reasons. Then it will be created inside `_RearchBootstrap`.

#### [bootstrap.dart](https://github.com/busslina/rearch-dart/blob/introducing-react-rearch/packages/react_rearch/lib/src/components/bootstrap.dart)
```dart
part of '../components.dart';

final capsuleContainer = CapsuleContainer();

final capsuleContainerContext = createContext<CapsuleContainer>();

ReactElement rearchBootstrap({
  required ReactDartComponentFactoryProxy2<Component2> child,
}) =>
    registerComponent2(_RearchBootstrap.new)({
      _PropsExtension.childField: child,
    });

extension _PropsExtension on _RearchBootstrap {
  static const childField = 'child';

  ReactDartComponentFactoryProxy2<Component2> get childComponent =>
      props[childField] as ReactDartComponentFactoryProxy2<Component2>;
}

class _RearchBootstrap extends Component2 {
  // final _capsuleContainer = CapsuleContainer();

  @override
  ReactNode render() {
    return capsuleContainerContext.Provider(
      {
        // 'value': _capsuleContainer,
        'value': capsuleContainer,
      },
      childComponent({}),
    );
  }
}
```

**Reply by busslina** [reply]
I just realized that rather create `CapsuleContainer` as a class field it can be setted as state with `initialState` getter.
